example/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"html/template"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/nicheinc/recaptcha"
+)
+
+var (
+	secretKey = flag.String("secret-key", "", "reCAPTCHA secret key")
+	siteKey   = flag.String("site-key", "", "reCAPTCHA site key")
+	hostnames = flag.String("hostname", "localhost", "Valid hostnames (comma separated)")
+	action    = flag.String("action", "example", "Action to be associated with reCAPTCHA tokens")
+	score     = flag.Float64("score", 0.5, "Minimum score threshold")
+	port      = flag.Int("port", 80, "Port to run example server on")
+
+	client recaptcha.Client
+)
+
+func main() {
+	flag.Parse()
+
+	client = recaptcha.NewClient(*secretKey)
+
+	http.HandleFunc("/", handler)
+	http.HandleFunc("/submit", submitHandler)
+	if err := http.ListenAndServe(fmt.Sprintf(":%d", *port), nil); err != nil {
+		log.Fatalf("Error serving requests: %s\n", err)
+	}
+}
+
+type data struct {
+	SiteKey string
+	Action  string
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	data := data{
+		SiteKey: *siteKey,
+		Action:  *action,
+	}
+
+	tmpl, err := template.New("template.html").ParseFiles("./template.html")
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("Error parsing html template: %s\n", err),
+			http.StatusInternalServerError,
+		)
+		return
+	}
+
+	if err := tmpl.Execute(w, data); err != nil {
+		http.Error(w,
+			fmt.Sprintf("Error executing template: %s", err),
+			http.StatusInternalServerError,
+		)
+		return
+	}
+}
+
+func submitHandler(w http.ResponseWriter, r *http.Request) {
+	token := r.FormValue("token")
+	if token == "" {
+		http.Error(w, "Form submission missing 'token' parameter", http.StatusBadRequest)
+		return
+	}
+	userIP, _, err := net.SplitHostPort(r.RemoteAddr)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("Error parsing remote addr: %s", err),
+			http.StatusBadRequest,
+		)
+		return
+	}
+
+	response, err := client.Fetch(context.Background(), token, userIP)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("Error making request to token verification endpoint: %s", err),
+			http.StatusInternalServerError,
+		)
+		return
+	}
+
+	out, err := json.MarshalIndent(response, "", "    ")
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("Error marshalling verification endpoint response: %s\n", err),
+			http.StatusInternalServerError,
+		)
+		return
+	}
+
+	if _, err := w.Write(out); err != nil {
```

Comment:
You could remove this whole `w.Write(out)` code block and only write to the ResponseWriter once, farther down in the handler; using `io.WriteString(w, string(out)+message)` instead of `io.WriteString(w, message)`